sourcegraph/sourcegraph
# Fix code insight live preview aria labels

Closes https://github.com/sourcegraph/sourcegraph/issues/42715
Closes https://github.com/sourcegraph/sourcegraph/issues/41687
 
## Background 

This PR marks up the creation UI pages in the following way (as it's presented on the schema below)

<img src="https://user-images.githubusercontent.com/18492575/194647350-cd60e376-281d-4c2d-b864-24e4a34e819d.png" width="400"/>

For example, I recorded a high-level walk-through of two versions of the creation UI live preview screen readers. 

| Search-based | Lang stats  |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/18492575/194648658-f4e5401e-fc7b-4963-8a07-ea084cc2b8ea.mov" /> |  <video src="https://user-images.githubusercontent.com/18492575/194648649-65814772-c040-44f9-89c9-6dbdcdcbb60d.mov" />  |



## Test plan
- Go through search-based, capture group, lang stats (compute powered) creation UI with a screen reader to the live preview card
- Make sure that initial (bluer chart and description message), loading and data states are pronounced correctly 
- Make sure that you check both the line and pie charts card.



## App preview:

- [Web](https://sg-web-vk-fix-live-preview-aria.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-idpwptukul.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
